firstvoices/backend/serializers/immersion_label_serializers.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import serializers
+
+from backend.models import DictionaryEntry
+from backend.models.immersion_labels import ImmersionLabel
+from backend.serializers.base_serializers import (
+    WritableSiteContentSerializer,
+    base_timestamp_fields,
+)
+from backend.serializers.dictionary_serializers import (
+    WritableRelatedDictionaryEntrySerializer,
+)
+from backend.serializers.fields import SiteHyperlinkedIdentityField
+
+
+class ImmersionLabelDetailSerializer(WritableSiteContentSerializer):
+    url = SiteHyperlinkedIdentityField(
+        view_name="api:immersionlabel-detail", read_only=True, lookup_field="key"
+    )
+    dictionary_entry = WritableRelatedDictionaryEntrySerializer(
+        required=True,
+        queryset=DictionaryEntry.objects.all(),
+    )
+    key = serializers.CharField(allow_blank=False, allow_null=False)
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        site = self.context["site"]
+        dictionary_entry = attrs.get("dictionary_entry")
+
+        if dictionary_entry and dictionary_entry.site != site:
```

Comment:
We have a validator for this-- you could define validators=[SameSite()] on the dictionary_entry field instead of validating it manually here.